programs/uxd/src/instructions/compute_depositories_targets.rs

```diff
@@ -0,0 +1,261 @@
+    )]
+    pub controller: AccountLoader<'info, Controller>,
+
+    /// #3 The first known active mercurial vault depository
+    #[account(
+        mut,
+        seeds = [
+            MERCURIAL_VAULT_DEPOSITORY_NAMESPACE,
+            mercurial_vault_depository.load()?.mercurial_vault.key().as_ref(),
+            mercurial_vault_depository.load()?.collateral_mint.as_ref()
+        ],
+        bump = mercurial_vault_depository.load()?.bump,
+        has_one = controller @UxdError::InvalidController,
+    )]
+    pub mercurial_vault_depository: AccountLoader<'info, MercurialVaultDepository>,
+
+    /// #4 The first known active credix lp depository
+    #[account(
+        mut,
+        seeds = [
+            CREDIX_LP_DEPOSITORY_NAMESPACE,
+            credix_lp_depository.load()?.credix_global_market_state.key().as_ref(),
+            credix_lp_depository.load()?.collateral_mint.as_ref()
+        ],
+        bump = credix_lp_depository.load()?.bump,
+        has_one = controller @UxdError::InvalidController,
+    )]
+    pub credix_lp_depository: AccountLoader<'info, CredixLpDepository>,
+}
+
+pub(crate) fn handler(ctx: Context<ComputeDepositoriesTargets>) -> Result<()> {
+    let controller = &ctx.accounts.controller.load()?;
+    let mercurial_vault_depository = &mut ctx.accounts.mercurial_vault_depository.load_mut()?;
+    let credix_lp_depository = &mut ctx.accounts.credix_lp_depository.load_mut()?;
+
+    // ---------------------------------------------------------------------
+    // -- Phase 1
+    // -- Read the desired weights for each depository on chain
+    // -- And generate a raw_target estimations that we can refine later
+    // ---------------------------------------------------------------------
+
+    // We want to balance the supply on weighted portions of the circulating supply for now
+    // This could be based on dynamic on-chain account values and liquidity later
+    let mercurial_vault_depository_weight_bps =
+        mercurial_vault_depository.redeemable_amount_under_management_weight_bps;
+    let credix_lp_depository_weight_bps =
+        credix_lp_depository.redeemable_amount_under_management_weight_bps;
+
+    // Compute raw target values based on weighted portions of circulating supply
+    let redeemable_circulating_supply =
+        checked_convert_u128_to_u64(controller.redeemable_circulating_supply)?;
+
+    let mercurial_vault_depository_raw_target = ctx.accounts.compute_raw_target(
+        redeemable_circulating_supply,
+        mercurial_vault_depository_weight_bps,
+    )?;
+    let credix_lp_depository_raw_target = ctx.accounts.compute_raw_target(
+        redeemable_circulating_supply,
+        credix_lp_depository_weight_bps,
+    )?;
+
+    // ---------------------------------------------------------------------
+    // -- Phase 2
+    // -- Using the raw_target and the depository cap:
+    // -- * Compute the overflow (raw target amount above the cap)
```

Comment:
nit: Comments of Phase 1 and Phase 2 use different style ("and" versus bullet points)